snap/info.go

```diff
@@ -226,3 +228,26 @@ func (app *AppInfo) ServiceFile() string {
 func (app *AppInfo) ServiceSocketFile() string {
 	return filepath.Join(dirs.SnapServicesDir, app.SecurityTag()+".socket")
 }
+
+// InfoWithSide returns the snap information for the installed snap with the given name and given side-info.
+func InfoWithSide(name string, si *SideInfo) (*Info, error) {
```

Comment:
Should we name this `ReadInfo`?  I think you suggested that before, and it seems quite nice in this package.